### PR TITLE
Add "Get Early Access" page

### PR DESCRIPTION
**Pull request by ceolson01**
## Description

This PR adds the _Get Early Access_ page

**New stuff** ✨

* Dependencies
  * Formik - for better forms
  * yup - for validation
* Get Early Access page

**Changes** 🏗

* Get early access links to new page
* Form in homepage hero pre-fills and redirects to new page

**Demo** 🎥 

Get Early Access Form

### CODE REVIEW

File: src/modules/core/components/Input/Input.jsx [hunk 1]
```diff
@@ -58,15 +61,18 @@ const Input = ({
...
 }: Props) => {
   const [styleProps, setStyleProps] = useState({});
 
   const classNames = className
-    ? `${getMainClasses(appearance, styles)} ${className}`
-    : getMainClasses(appearance, styles);
+    ? `${getMainClasses(appearance, styles, {
+        hasError: !!error,
+      })} ${className}`
+    : getMainClasses(appearance, styles, { hasError: !!error });
```
Comment (ryanchristo) [inline src/modules/core/components/Input/Input.jsx]: Nice use of state. :+1:

File: src/modules/core/components/Input/Input.jsx [hunk 2]
```diff
@@ -109,6 +114,12 @@ const Input = ({
         value={value}
         {...rest}
       />
+      {showErrorText && (
+        // Always render this div so the UI doesn't shift once an error message exists
```
Comment (ryanchristo) [inline src/modules/core/components/Input/Input.jsx:117]: Well commented. `showErrorText` is a nice addition too.

File: src/modules/core/components/Textarea/Textarea.module.css [hunk 3]
```diff
@@ -0,0 +1,32 @@
+
+.main {
+  border: 1px solid var(--grey-2);
+  color: var(--text-blue);
+  margin-top: 8px;
+  /* Negative margin to match `Input`s */
```
Comment (ryanchristo) [inline src/modules/core/components/Textarea/Textarea.module.css:6]: :ok_hand:

File: src/modules/pages/components/Website/EarlyAccess/EarlyAccess.jsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    state?: { email?: string },
+  },
+|};
+
+const displayName = 'pages.Website.EarlyAccess';
+
+const EarlyAccess = ({ location }: Props) => {
+  const email =
+    location && location.state && location.state.email
+      ? location.state.email
```
Comment (ryanchristo) [inline src/modules/pages/components/Website/EarlyAccess/EarlyAccess.jsx:59]: This isn't working for me; `location` is always `undefined`. Do we need to call the `withLocation` method to make `location` accessible?

File: src/modules/pages/components/Website/HomePage/Hero/Hero.jsx [hunk 5]
```diff
@@ -45,10 +46,8 @@ const Hero = () => {
   const style = headerHeight ? { paddingTop: `${headerHeight}px` } : {};
 
   const handleSubmit = useCallback((email: string) => {
-    const endpoint = `${PAGE_GET_EARLY_ACCESS}?email=${email}`;
-    if (window !== 'undefined') {
-      window.open(endpoint, '_blank');
-    }
+    // $FlowFixMe - flow type state below
+    navigate(PAGE_GET_EARLY_ACCESS, { state: { email } });
```
Comment (ryanchristo) [inline src/modules/pages/components/Website/HomePage/Hero/Hero.jsx]: This looks right but as mentioned above, `location` is always `undefined` in `EarlyAccess`.

**Review by ryanchristo — CHANGES_REQUESTED** [review]
Hmmm... I'm not able to get my email address to transfer over from one page to the other. Maybe something changed since the demo reel? If using `location` in `EarlyAccess`, do we need to call the `withLocation` method at some point to make `location` accessible?

File: src/modules/pages/components/Website/EarlyAccess/EarlyAccess.jsx (hunk 4 above)
Comment (ceolson01) [reply src/modules/pages/components/Website/EarlyAccess/EarlyAccess.jsx:59]: I've adjusted the props to pass through from gatsby pages. All good now

**Review by ryanchristo — APPROVED** [review]
Nice! Works now! Approved. :+1:

**Pull request merged**